### PR TITLE
Syntax highlighting in docs

### LINKED ISSUE #1781: Documentation Pages Missing
Hello!

I'm not sure if this is the correct place to bring this up, but the documentation for Jdaviz does not include a section of who/where to ask for assistance (https://jdaviz.readthedocs.io/en/latest/index.html).

I'm trying to get Jdaviz running and I'm fairly certain instructions are missing on the documentation. There are the installation instructions, and I went through the conda route. However after that there are no instructions on how to launch the GUI or how to get the jdaviz command working on the command line (it does not for me). 

On GitHub (https://github.com/spacetelescope/jdaviz) there is a small quick start section, that links to a quick start page that does not exist (https://jdaviz.readthedocs.io/en/latest/quickstart.html). 

So in summary, it looks like the documentation needs:
1) A quick start page to help people get it up and running
2) Somewhere to ask for help. Because I cannot get it running but I do not know who to ask.

Thank you very much and I look forwards to using jdaviz!

### PR DESCRIPTION
**Pull request by bmorris3**
This PR makes some revisions to the sphinx docs and docstrings. PR outline: (1) improves readability of code blocks, (2) fixes broken internal references within the sphinx docs (in both `.rst` files and several docstrings, and (3) fixes a broken link in the README.

Fix #1781

### Preview

Preview these changes in this branch's [RTD build](https://jdaviz--1768.org.readthedocs.build/en/1768/). Prepare for colorful code 🌈

### Details

1. As I have been reading the docs to come up to speed on `jdaviz`, I noticed a convention to use `::` before a code block in the `docs/*.rst` files. This works fine, but if you replace it with:
```diff
- some code::

+ some code: 
+ 
+ .. code-block:: <syntax>
```
you get syntax highlighting within the code blocks. This makes the Python blocks a bit easier to read, and distinguishes the Python from bash.

2. Our narrative docs are just a few characters away from successfully cross-referencing the API documentation for each class and method. I've made revisions in this PR that fix some broken cross-references throughout our documentation.

3. The link in [this section of the README](https://github.com/spacetelescope/jdaviz/#as-a-web-application) is broken because that docs page doesn't exist anymore. I've changed the link in this PR.

#### Further/future work

I mentioned in #1767 that there are still several missing docstrings, even if this PR gets merged. We should probably write those, too!

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Doc only change does not need a change log entry. The doc changes would appear to user anyway and does not change the actual package functionality. Please remove the change log changes and squash it out. Thanks!

File: README.rst [hunk 1]
```diff
@@ -94,7 +103,9 @@ To learn more about the various ``jdaviz`` application configurations and loadin
 
 ``jdaviz`` also provides a directory of `sample notebooks <https://jdaviz.readthedocs.io/en/latest/sample_notebooks.html>`_ to test the application, located in the ``notebooks`` sub-directory
 of the git repository.  ``CubevizExample.ipynb`` is provided as an example that loads a SDSS MaNGA IFU data cube with the
```
[the remark this answers is not in the document]
Comment (bmorris3) [reply README.rst:105]: Good catch!

File: docs/specviz2d/plugins.rst [hunk 2]
```diff
@@ -69,7 +69,7 @@ adjust plotting options), open the "Export Trace" panel, choose a label for the
 and click "Create".  Note that this step is not required to create an extraction with simple
 workflows.
 
-Trace objects created outside of jdaviz can be loaded into the app via :meth:`~jdaviz.configs.specviz2d.helper.Specviz2d.load_trace`:
+Trace objects created outside of jdaviz can be loaded into the app via :py:meth:`~jdaviz.configs.specviz2d.helper.Specviz2d.load_trace`:
```
[the remark this answers is not in the document]
Comment (bmorris3) [reply docs/specviz2d/plugins.rst:72]: Thanks, honestly not sure how I missed these. Fixing them now, too. (Maybe PyCharm's Find feature isn't as thorough as I expect it to be?)

File: docs/specviz/import_data.rst [hunk 3]
```diff
@@ -62,14 +66,16 @@ The example below loads a FITS file into Specviz::
     specviz.show()
 
 You can also pass the path to a file that `~specutils.Spectrum1D` understands directly to the
-:meth:`jdaviz.configs.specviz.helper.Specviz.load_spectrum` method::
+:py:meth:`~jdaviz.configs.specviz.helper.Specviz.load_spectrum` method::
```
Comment (pllim) [inline docs/specviz/import_data.rst:69]:
```suggestion
:py:meth:`~jdaviz.configs.specviz.helper.Specviz.load_spectrum` method:

.. code-block:: python
```

**Review by pllim — COMMENTED** [review]
Almost there... 😸

File: README.rst [hunk 4]
```diff
@@ -1,10 +1,11 @@
 .. image:: docs/logos/jdaviz.svg
-   :width: 400
-   :alt: Jdaviz logo
+    :width: 400
+    :alt: Jdaviz logo
+    :align: center
```
[the remark this answers is not in the document]
Comment (bmorris3) [reply README.rst:4]: Interestingly, it does, but not always? This is a feature I've wanted for a long time in GH renders of rst docs, and [it works in some repos](https://github.com/github/markup/issues/163#issuecomment-1282676832) but not this one... not sure why?

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**